sound/soc/sof/core.c

```diff
@@ -215,16 +216,31 @@ int snd_sof_create_page_table(struct snd_sof_dev *sdev,
 		 */
 		u32 idx = (5 * i) >> 1;
 		u32 pfn = snd_sgbuf_get_addr(dmab, i * PAGE_SIZE) >> PAGE_SHIFT;
-		u32 *pg_table;
+		u8 *pg_table;
 
 		dev_vdbg(sdev->dev, "pfn i %i idx %d pfn %x\n", i, idx, pfn);
 
-		pg_table = (u32 *)(page_table + idx);
+		pg_table = (u8 *)(page_table + idx);
 
+		/*
```

Comment:
The commit is super nice! One thing if you can do is: explain why the compression is needed for readers who will try to understand the code later.